pkg/flow/internal/controller/loader_test.go

```diff
@@ -205,7 +205,7 @@ func TestScopeWithFailingComponent(t *testing.T) {
 	diags := applyFromContent(t, l, []byte(testFile))
 	require.Error(t, diags.ErrorOrNil())
 	require.Len(t, diags, 1)
-	require.True(t, strings.Contains(diags.Error(), "Failed to build component: decoding River: missing required attribute \"frequency\""))
+	require.True(t, strings.Contains(diags.Error(), `unrecognized attribute name "frequenc"`))
```

Comment:
This needed to be changed because the new evaluation order will report unrecognized attributes first over missing a required attribute.